DEVELOPER_GUIDE.md

```diff
@@ -74,6 +84,74 @@ This is typically created in PyCharm IDE by visiting the `Python Interpreter`, s
 `
 In order to run tests within the PyCharm IDE, ensure the `Python Integrated Tools` / `Testing` / `Default Test Runner` is set to `pytest`.
 
+## Setting Up a Local OpenSearch Cluster For OSB Development (Optional)
+
+### OpenSearch Installation  
+
+Download the latest release of OpenSearch from https://opensearch.org/downloads.html. If you are using WSL, make sure to download it into your `/home/<user>` directory instead of `/mnt/c`. 
+```
+wget https://artifacts.opensearch.org/releases/bundle/opensearch/<x.x.x>/opensearch-<x.x.x>-linux-x64.tar.gz
+tar -xf opensearch-x.x.x-linux-x64.tar.gz
+cd opensearch-x.x.x
+```
+NOTE: Have Docker running in the background for the next steps. Refer to the installation instructions [here](https://docs.docker.com/compose/install/).
+
+### OpenSearch Cluster setup
+
+Add the following settings to the `opensearch.yml` file under the config directory 
+```
+vim config/opensearch.yml
+```
+```
+#
+discovery.type: single-node 
+plugins.security.disabled: true 
```

Comment:
This should not be necessary, if certificate checking is eliminated. See https://github.com/opensearch-project/documentation-website/pull/5177.